File: frontend/Frontend_FinalCapstone/src/app/models/promotion-preview.dto.ts
/**
 * PromotionPreviewDTO - Preview of discount calculation
 * Used for validation before payment
 * 
 * ⚠️ IMPORTANT: Backend Java field "isValid" becomes "valid" in JSON
 * due to Jackson's boolean serialization (removes "is" prefix)
 */
export interface PromotionPreviewDTO {
  valid: boolean;              // ✅ Backend sends "valid", not "isValid"
  errorMessage?: string | null;
  originalAmount: number;
  discountAmount: number;
  finalAmount: number;
  promotionCode?: string;
  promotionName?: string;
  description?: string | null;
  discountType?: string | null;    // "PERCENTAGE" or "FIXED"
  discountValue?: number | null;
}

File: backend/Final_Capstone/src/main/java/com/Cybersoft/Final_Capstone/dto/PromotionPreviewDTO.java
package com.Cybersoft.Final_Capstone.dto;

import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.math.BigDecimal;

/**
 * DTO for previewing promotion discount without consuming the promotion
 */
@Data
@NoArgsConstructor
@AllArgsConstructor
@Builder
public class PromotionPreviewDTO {
    private String promotionCode;
    private String promotionName;
    private String description;
    private BigDecimal originalAmount;
    private BigDecimal discountAmount;
    private BigDecimal finalAmount;
    private String discountType; // PERCENTAGE or FIXED
    private BigDecimal discountValue; // e.g., 10 for 10% or $10
    private boolean isValid;
    private String errorMessage; // if not valid
}




